django/contrib/postgres/constraints.py

```diff
@@ -124,9 +120,19 @@ def remove_sql(self, model, schema_editor):
         )
 
     def check_supported(self, schema_editor):
-        if self.include and not schema_editor.connection.features.supports_covering_gist_indexes:
+        if (
+            self.include and self.index_type.lower() == 'gist' and
+            not schema_editor.connection.features.supports_covering_gist_indexes
+        ):
+            raise NotSupportedError(
+                'Covering exclusion constraints using a GiST index requires PostgreSQL 12+.'
+            )
+        if (
+            self.include and self.index_type.lower() == 'spgist' and
+            not schema_editor.connection.features.supports_covering_spgist_indexes
+        ):
```

Comment:
Can you group both of these under `if self.include`?

Reply:
I wasn't too sure about this point...

Yes it avoids checking `self.include`'s truthiness twice, but:

- It makes each check less self contained. When we drop support for PostgreSQL 11 do we recombine?
- Also, what if another check is added unrelated to `self.include`. It'd have to go before the first if statement.

I love early returns in general, but it's not clear to me that they're always appropriate.

Reply:
reverted